src/scripts/main.js

```diff
@@ -1 +1,73 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.removeEventListener('click', resolve);
+  };
+
+  document.addEventListener('click', handler);
+
+  setTimeout(reject, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.removeEventListener('click', resolve);
+    document.removeEventListener('contextmenu', resolve);
+  };
+
+  document.addEventListener('contextmenu', handler);
+  document.addEventListener('click', handler);
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let left = false;
+  let right = false;
+
+  const leftClick = () => {
+    left = true;
+
+    if (right) {
+      resolve();
+      document.removeEventListener('click', leftClick);
```

Comment:
Bug: Here, you are removing the 'click' event listener with the 'leftClick' function. However, you should also remove the 'contextmenu' event listener with the 'leftClick' function as it was added with the same handler.